pynest/examples/eprop_plasticity/eprop_supervised_classification_evidence-accumulation.py

```diff
@@ -325,49 +325,39 @@ def calculate_glorot_dist(fan_in, fan_out):
 weights_out_rec = np.array(np.random.randn(n_rec, n_out), dtype=dtype_weights)
 
 params_common_syn_eprop = {
-    "adam_beta1": 0.9,  # exponential decay rate for 1st moment estimate of Adam optimizer
-    "adam_beta2": 0.999,  # exponential decay rate for 2nd moment raw estimate of Adam optimizer
-    "adam_epsilon": 1e-8,  # small numerical stabilization constant of Adam optimizer
+    "optimizer": {
+        "type": "adam",  # algorithm to optimize the weights
+        "batch_size": n_batch,
+        "beta_1": 0.9,  # exponential decay rate for 1st moment estimate of Adam optimizer
+        "beta_2": 0.999,  # exponential decay rate for 2nd moment raw estimate of Adam optimizer
+        "epsilon": 1e-8,  # small numerical stabilization constant of Adam optimizer
+        "eta": 5e-3,  # learning rate
+        "Wmin": -100.0,  # pA, minimal limit of the synaptic weights
+        "Wmax": 100.0,  # pA, maximal limit of the synaptic weights
+    },
     "average_gradient": True,  # if True, average the gradient over the learning window
-    "batch_size": n_batch,
-    "optimizer": "adam",  # algorithm to optimize the weights
     "weight_recorder": wr,
 }
 
 params_syn_in = {
     "synapse_model": "eprop_synapse",
-    "adam_m": 0.0,  # initial 1st moment estimate m of Adam optimizer
-    "adam_v": 0.0,  # initial 2nd moment raw estimate v of Adam optimizer
     "delay": duration["step"],  # ms, dendritic delay
```

Comment:
```suggestion
    "synapse_model": "eprop_synapse",
    "adam_m": 0.0,  # initial 1st moment estimate m of Adam optimizer
    "adam_v": 0.0,  # initial 2nd moment raw estimate v of Adam optimizer
    "delay": duration["step"],  # ms, dendritic delay
```
Here, in the dictionary together with other parameters it might make sense to keep `adam_m` instead of `m` since it is not a subdictionary. I would just keep the user-facing name `adam_m` and in the c++ code `m`. This would unfortunately require renaming `names::m` to `names::adam_m`.  The same applies to `adam_v`.  If you prefer `m`, I can also add this change in a later commit.

Reply:
This will not work. You need to pass optimizer state variables inside the optimizer dictionary, but you can do that only for actually existing synapses like this:
```python
n = nest.Create('eprop_iaf_psc_delta')
nest.SetDefaults('eprop_synapse', {'optimizer': {'type': 'adam'}})
nest.Connect(n, n, syn_spec={'synapse_model': 'eprop_synapse', 'delay': nest.resolution})
c = nest.GetConnections()
c.set({'optimizer': {'m': 1, 'v': 0.5}})
```
This is a consequence of not having an optimizer object in the default connection.

Reply:
I see, thanks for the explanation. Then, maybe in a later commit we can add a hint on that to the tutorial.